File: Data Structures & Algorithms/string-encode-and-decode/submission-0.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        curr = 0
        self.pos = []
        res = ""
        for s in strs:
            res += s
            self.pos.append((curr, curr + len(s) - 1))
            curr += len(s)
        return res
    def decode(self, s: str) -> List[str]:
        res = []
        for l, r in self.pos:
            res.append(s[l:r + 1])
        return res

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int res = 0;
        int curr = INT_MAX;
        for (int price: prices) {
            if (curr == INT_MAX) {
                curr = price;
                continue;
            }
            if (price > curr) res = max(res, price - curr);
            else if (price < curr) curr = price;

        }
        return res;
    }
};
